tests/validation/weight_test.go

```diff
@@ -15,27 +15,25 @@ func weightTests(t *testing.T, when spec.G, it spec.S) {
 	var service testutil.TestService
 	var stagedService testutil.TestService
 
-	it.Before(func() {
-		service.Create(t, "ibuildthecloud/demo:v1")
-		stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
-	})
-
-	it.After(func() {
-		service.Remove()
-		stagedService.Remove()
-	})
-
 	when("a staged service incrementally rolls out weight", func() {
+		it.Before(func() {
+			service.Create(t, "--weight", "100", "ibuildthecloud/demo:v1")
+			stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
+		})
+
+		it.After(func() {
+			service.Remove()
+			stagedService.Remove()
+		})
 		it("should slowly increase weight on the staged service and leave service weight unchanged", func() {
+			assert.Equal(t, 100, service.GetComputedWeight())
 			// The time from rollout to obtaining the current weight, without Sleep, is 2 seconds.
-			// Sleeping 8 seconds here with a rollout-interval of 4 seconds to guarantee 2 rollout ticks, plus the initial tick, with 2 seconds to spare.
-			stagedService.WeightWithoutWaiting(80, "--duration=1m")
+			// Sleeping 9 seconds to guarantee 5 rollout ticks with 1 second to spare since the default tick interval is 2 seconds.
+			stagedService.WeightWithoutWaiting(60, "--duration=1m")
 			time.Sleep(8 * time.Second)
-
-			stagedWeightAfter10Seconds := stagedService.GetCurrentWeight()
-			serviceWeightAfter10Seconds := service.GetCurrentWeight()
-			assert.Equal(t, 30, stagedWeightAfter10Seconds)
-			assert.Equal(t, 100, serviceWeightAfter10Seconds)
+			stagedComputedWeightAfter10Seconds, serviceComputedWeightAfter10Seconds := stagedService.GetComputedWeight(), service.GetComputedWeight()
+			assert.Equal(t, 20, stagedComputedWeightAfter10Seconds)
```

Comment:
Does this actually work ? I'm worried that as I change the rollout config this will break, if we add a magic +- 50% scaler thing the ticks won't be equal anymore. Guess we can change later, but I think it would be better to do a > 10 < 30 or something. We can figure out later when I break it :)

Reply:
I considered that. This did seem to work after I set time to 9, but when time was 8 I noticed weight would be at 17 about 30% of the time, and time at 10 the weight was 25 about 30% of the time. So this could've definitely been flaky. 
Fixed to your suggestion, and will update as needed based on your rollout config changes.